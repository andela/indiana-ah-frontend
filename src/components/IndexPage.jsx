import React, { Component } from 'react';
import Modal from './common/Modal.jsx';
import LoginContainer from './LoginFormContainer.jsx';

class IndexPage extends Component {
  state = {
    modalIsOpen: false,
    emailValue: '',
    passwordValue: '',
    openLoginForm: true,
    openRegisterForm: false
  };

  handleSubmit = (e) => {
    e.preventDefault();
  };

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  openModal = () => {
    this.setState(() => ({ modalIsOpen: true }));
  };

  closeModal = () => {
    this.setState(() => ({ modalIsOpen: false }));
  };

  openRegisterForm = () => {
    this.setState({
      openRegisterForm: true,
      openLoginForm: false
    });
  };

  openLoginForm = () => {
    this.setState({
      openRegisterForm: false,
      openLoginForm: true
    });
  };

  render() {
    const { openLoginForm } = this.state;
    return (
      <div className="container">
        <h2>Welcome to Authors Haven!</h2>
        {/* This is how to use the modal */}
        <button onClick={this.openModal}>Open Modal</button>
        <Modal
          modalIsOpen={this.state.modalIsOpen}
          closeModal={this.closeModal}
          body={
            openLoginForm ? (
              <LoginContainer
                loginForm
                handleChange={this.handleChange}
                closeModal={this.closeModal}
                openLoginForm={this.openLoginForm}
                openRegisterForm={this.openRegisterForm}
                handleSubmit={this.handleSubmit}
              />
            ) : (
              <LoginContainer
                registerForm
                handleChange={this.handleChange}
                closeModal={this.closeModal}
                openLoginForm={this.openLoginForm}
                openRegisterForm={this.openRegisterForm}
                handleSubmit={this.handleSubmit}
              />
            )
          }
        />
      </div>
    );
  }
}

export default IndexPage;
