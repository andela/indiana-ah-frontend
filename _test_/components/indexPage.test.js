import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import IndexPage from '../../src/components/IndexPage.jsx';

describe('Custom Footer component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <IndexPage />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the Index Page', () => {
    const indexPage = shallow(<IndexPage />);
    expect(indexPage.find('.container')).toBeDefined();
    const h2 = indexPage.find('h2');
    expect(h2.length).toEqual(1);
    expect(h2.text()).toEqual('Welcome to Authors Haven!');
  });
});
