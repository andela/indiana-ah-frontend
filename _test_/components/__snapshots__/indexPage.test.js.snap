// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page component renders correctly 1`] = `
Array [
  <div
    className="container"
  >
    <div
      className="banner row"
    >
      <div
        className="d-flex align-items-center col"
      >
        <div>
          <h1
            className="w-80"
          >
            Share Your Thoughts With The World
          </h1>
          <p
            className="banner-p"
          >
            There is no limit to what you can achieve with a pen.. Write, the world is ready to read.
          </p>
          <button
            className="sc-bdVaJa bsDjTF"
          >
            Get Started
          </button>
        </div>
      </div>
      <div
        className="col"
      >
        <span
          className="isvg loading"
        />
      </div>
    </div>
  </div>,
  <section
    className="container-fluid reviews"
  >
    <div
      className="justify-content-center row"
    >
      <div
        className="col-md-8"
      >
        <h2
          className="text-center"
        >
          A community of like-minded authors fostering inspiration and innovation by leveraging the modern web.
        </h2>
      </div>
    </div>
    <div
      className="text-center row"
    >
      <div
        className="col-md-4"
      >
        <p>
          Author’s haven indeed lives up to its name.
          It has been a source of succor and comfort to me.
          In here I have found individuals of like mind as I and
          I have also gained inspiration and motivation.
          The author’s haven team are also very efficient, professional 
          and prompt with their responses.
        </p>
        <div
          className="quotes"
        >
          <span
            className="isvg loading"
          />
        </div>
        <p
          className="reviews-name"
        >
          Tiku Okoye
        </p>
        <p>
          California
        </p>
      </div>
      <div
        className="col-md-4"
      >
        <p>
          Author’s haven indeed lives up to its name.
          It has been a source of succor and comfort to me.
          In here I have found individuals of like mind as I and
          I have also gained inspiration and motivation.
          The author’s haven team are also very efficient, professional 
          and prompt with their responses.
        </p>
        <div
          className="quotes"
        >
          <span
            className="isvg loading"
          />
        </div>
        <p
          className="reviews-name"
        >
          Dozie Nwoga
        </p>
        <p>
          Anambra
        </p>
      </div>
      <div
        className="col-md-4"
      >
        <p>
          Author’s haven indeed lives up to its name.
          It has been a source of succor and comfort to me.
          In here I have found individuals of like mind as I and
          I have also gained inspiration and motivation.
          The author’s haven team are also very efficient, professional 
          and prompt with their responses.
        </p>
        <div
          className="quotes"
        >
          <span
            className="isvg loading"
          />
        </div>
        <p
          className="reviews-name"
        >
          Ozone Ezenwa
        </p>
        <p>
          Hawaii
        </p>
      </div>
    </div>
  </section>,
  <section
    className="container"
  >
    <div
      className="banner row"
    >
      <div
        className="col-md-7"
      >
        <span
          className="isvg loading"
        />
      </div>
      <div
        className="d-flex align-items-center col"
      >
        <div>
          <h2>
            Are you ready to put down those wonderful ideas?
          </h2>
          <p
            className="banner-p2"
          >
            Get Started for free today.
          </p>
          <button
            className="sc-bdVaJa iQmcrA"
          >
            Get Started
          </button>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="container-fluid"
  >
    <div
      className="subscription row"
    >
      <div
        className="col-md-1"
      />
      <div
        className="position-relative col-md-4"
      >
        <div
          className="w-80"
        >
          <h2>
            Subscribe To Email List
          </h2>
          <p
            className="banner-p2"
          >
            Interdum consectetur libero id faucibus nisl. Risus pretium quam vulputate dignissim suspendisse in
          </p>
        </div>
        <div
          className="sub-form"
        >
          <div
            className="input-div"
          >
            <input
              className="sc-bwzfXH jLxEfk"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
            <small
              className="text-danger ml-3 form-text"
            />
          </div>
          <button
            className="sc-bdVaJa gbqfHe"
          >
            Get Started
          </button>
        </div>
      </div>
      <div
        className="img-div p-0 col"
      >
        <img
          className="banner-img"
          src="../src/assets/images/index_img.jpg"
        />
      </div>
    </div>
  </section>,
  <div
    className="footer"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-4 col-md-5 col-sm-4 col-3 d-none d-sm-block pl-0 align-self-center col"
        >
          <div
            className="d-flex"
          >
            <span
              className="isvg loading feather"
            />
            <span
              className="isvg loading d-block app-logo"
            />
          </div>
        </div>
        <div
          className="d-flex align-items-center flex-column col"
        >
          <h2
            className="mb-5"
          >
            Help
          </h2>
          <a
            href="/"
            onClick={[Function]}
          >
            <h4
              className="mb-4"
            >
              Profile
            </h4>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <h4>
              Blog
            </h4>
          </a>
        </div>
        <div
          className="d-flex align-items-center flex-column col"
        >
          <h2
            className="mb-5"
          >
            Contact Us
          </h2>
          <a
            href="/"
            onClick={[Function]}
          >
            <h4
              className="mb-4"
            >
              Faqs
            </h4>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <h4>
              Contact us
            </h4>
          </a>
        </div>
        <div
          className="d-flex align-items-center flex-column col"
        >
          <h2
            className="mb-4"
          >
            Social Media
          </h2>
          <div
            className="d-flex justify-evenly w-75 mb-4"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <span
                className="isvg loading mr-5 d-block logo2"
              />
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <span
                className="isvg loading d-block logo2"
              />
            </a>
          </div>
          <div
            className="d-flex justify-content-between w-75"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <h4>
                Facebook
              </h4>
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              <h4>
                Twitter
              </h4>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
